ui/user/src/lib/services/chat/operations.ts

```diff
@@ -1379,3 +1420,111 @@ export async function bootstrapLogin(token: string) {
+	return response.items ?? [];
+}
+
+export async function getSingleOrRemoteMcpServer(
+	id: string,
+	opts?: { fetch?: Fetcher }
+): Promise<MCPCatalogServer> {
+	const response = (await doGet(`/mcp-servers/${id}`, opts)) as MCPCatalogServer;
+	return response;
+}
+
+export async function createSingleOrRemoteMcpServer(server: {
+	catalogEntryID?: string;
+	manifest?: {
+		url?: string;
+	};
+}): Promise<MCPCatalogServer> {
+	const response = (await doPost('/mcp-servers', server)) as MCPCatalogServer;
+	return response;
+}
+
+export async function updateSingleOrRemoteMcpServer(
+	id: string,
+	server: MCPServer
+): Promise<MCPCatalogServer> {
+	const response = (await doPut(`/mcp-servers/${id}`, server)) as MCPCatalogServer;
+	return response;
+}
+
+export async function deleteSingleOrRemoteMcpServer(id: string): Promise<void> {
+	await doDelete(`/mcp-servers/${id}`);
+}
+
+export async function configureSingleOrRemoteMcpServer(
+	id: string,
+	envs: Record<string, string>
+): Promise<void> {
+	await doPost(`/mcp-servers/${id}/configure`, envs);
+}
+
+export async function deconfigureSingleOrRemoteMcpServer(id: string): Promise<void> {
+	await doPost(`/mcp-servers/${id}/deconfigure`, {});
+}
+
+export async function revealSingleOrRemoteMcpServer(id: string): Promise<Record<string, string>> {
+	return doPost(`/mcp-servers/${id}/reveal`, {}) as Promise<Record<string, string>>;
+}
+
+export async function listSingleOrRemoteMcpServerTools(id: string): Promise<MCPServerTool[]> {
+	const response = (await doGet(`/mcp-servers/${id}/tools`, {
+		dontLogErrors: true
+	})) as ItemsResponse<MCPServerTool>;
+	return response.items ?? [];
+}
+
+export async function listSingleOrRemoteMcpServerPrompts(id: string): Promise<MCPServerPrompt[]> {
+	const response = (await doGet(`/mcp-servers/${id}/prompts`, {
+		dontLogErrors: true
+	})) as ItemsResponse<MCPServerPrompt>;
+	return response.items ?? [];
+}
+
+export async function listSingleOrRemoteMcpServerResources(
+	id: string
+): Promise<McpServerResource[]> {
+	const response = (await doGet(`/mcp-servers/${id}/resources`, {
+		dontLogErrors: true
+	})) as ItemsResponse<McpServerResource>;
+	return response.items ?? [];
+}
+
+export async function listMcpServerInstances(opts?: {
+	fetch?: Fetcher;
+}): Promise<MCPServerInstance[]> {
+	const response = (await doGet('/mcp-server-instances', opts)) as ItemsResponse<MCPServerInstance>;
+	return response.items ?? [];
+}
+
+export async function getMcpServerInstance(
+	id: string,
+	opts?: { fetch?: Fetcher }
+): Promise<MCPServerInstance> {
+	const response = (await doGet(`/mcp-server-instances/${id}`, opts)) as MCPServerInstance;
+	return response;
+}
+
+export async function createMcpServerInstance({
+	mcpServerID,
+	mcpCatalogID
+}: {
+	mcpServerID?: string;
+	mcpCatalogID?: string;
+}): Promise<MCPServerInstance> {
+	const response = (await doPost('/mcp-server-instances', {
+		mcpServerID,
+		mcpCatalogID
+	})) as MCPServerInstance;
+	return response;
+}
```

Comment:
The `mcpCatalogID` will be ignored here. We auto-populate that on the backend if the MCPServer pointed to by `mcpServerID` belongs to a catalog.